flict/flictlib/arbiter.py

```diff
@@ -0,0 +1,220 @@
+        # Let's loop through all the other licenses in the combined work
+        #
+        # TODO: sort all_licenses with repsect to license preferences
+        for lic in all_licenses:
+            compatible = self._combined_work_compatible(lic, package_info, dep_infos)
+            if compatible:
+                outbound_licenses.add(lic)
+
+        return list(outbound_licenses)
+
+    def _package_info(self, package, licenses):
+        compats = self.verify_package(package, licenses)
+
+        return {
+            'name': package['name'],
+            'license': package['license'],
+            'licenses_to_check': list(licenses),
+            'version': package['version'],
+            'description': package.get('description', ""),
+            'compatibility': compats
+        }
+
+    def verify(self, project, licenses=None):
+        """Verifies a project's license to a list of outbounds and returns the
+        compatibility between the liceses.
+             Parameters:
+                 project: the project (with its packages and their licenses) to check for compatibility
+                 licenses: the licenses to check the package's license against
+        """
+
+        start_time = timestamp()
+
+        project_name = project['project_name']
+
+        package_infos = []
+        all_licenses = set()
+
+        for package in project['packages']:
+            license_expression = Project.combined_work_license(package)
+
+            if licenses is None:
+                licenses = self.license_compatibility.licenses(license_expression)
+
+            all_licenses.update(licenses)
+
+            package_info = self._package_info(package, licenses)
+
+            dep_infos = []
+            for dep in package.get('dependencies', []):
+                dep_info = self._package_info(dep, licenses)
+                dep_infos.append(dep_info)
+            package_info['dependencies'] = dep_infos
+
+            outbound_licenses = self._top_package_license(package['license'], licenses, package_info, dep_infos)
+            package_info['outbound_licenses'] = outbound_licenses
+
+            package_info['outbound_licenses_aliased'] = []
+            for lic in outbound_licenses:
+                package_info['outbound_licenses_aliased'].append(self.license_compatibility.replace_aliases(lic))
+
+            # Chose outbound
+            chosen_alias = self.license_compatibility.chose_license(package_info['outbound_licenses_aliased'])
+            if chosen_alias is None:
+                chosen_license = None
+                package_info['outbound_license'] = None
+                package_info['outbound_license_aliased'] = None
+            else:
+                index = package_info['outbound_licenses_aliased'].index(chosen_alias)
+                chosen_license = outbound_licenses[index]
+                package_info['outbound_license'] = chosen_license
+                package_info['outbound_license_aliased'] = chosen_alias
```

Comment:
This part is still confusing. In the previous lines you get outbound licenses and outbound licenses aliased. None of them matter because you override them either way in this section. Why write temporary data into `package_info`?

Reply:
I have added some comments on `outbound_license` vs `outbound_licenses` (singular vs plural) hoping that they make it easier to follow

Reply:
The idea is to give a compliance person a chance to see which license was chosen as outbound and from what licenses that was chosen from,

Reply:
Ah my sloppy reading strikes again. Maybe that naming could be improved in the future.